File: core-banking-prototype-laravel-main/core-banking-prototype-laravel-main/sdks/python/finaegis/resources/transfers.py
"""
Transfers resource for the FinAegis SDK
"""

from typing import Optional
from ..types import Transfer
from .base import BaseResource


class TransfersResource(BaseResource):
    """Manage transfers in the FinAegis platform."""
    
    def create(
        self,
        from_account: str,
        to_account: str,
        amount: int,
        asset_code: str = 'USD',
        reference: Optional[str] = None,
        workflow_enabled: bool = True
    ) -> Transfer:
        """
        Create a new transfer.
        
        Args:
            from_account: Source account UUID
            to_account: Destination account UUID
            amount: Amount in cents
            asset_code: Asset code (default: USD)
            reference: Optional reference for the transfer
            workflow_enabled: Whether to enable workflow processing
            
        Returns:
            Transfer object
        """
        data = {
            'from_account': from_account,
            'to_account': to_account,
            'amount': amount,
            'asset_code': asset_code,
            'workflow_enabled': workflow_enabled
        }
        if reference:
            data['reference'] = reference
            
        response = self._post('/transfers', data)
        return Transfer.from_dict(response['data'])
    
    def get(self, uuid: str) -> Transfer:
        """
        Get transfer details.
        
        Args:
            uuid: Transfer UUID
            
        Returns:
            Transfer object
        """
        response = self._get(f'/transfers/{uuid}')
        return Transfer.from_dict(response['data'])

File: core-banking-prototype-laravel-main/core-banking-prototype-laravel-main/sdks/php/src/Resources/Transfers.php
<?php

namespace FinAegis\Resources;

use FinAegis\Models\PaginatedResponse;
use FinAegis\Models\Transfer;

class Transfers extends BaseResource
{
    /**
     * Create a new transfer.
     *
     * @param  string  $fromAccount  Source account UUID
     * @param  string  $toAccount  Destination account UUID
     * @param  float  $amount  Amount in cents
     * @param  string  $assetCode  Asset code
     * @param  string|null  $reference  Optional reference
     */
    public function create(
        string $fromAccount,
        string $toAccount,
        float $amount,
        string $assetCode,
        ?string $reference = null
    ): Transfer {
        $data = [
            'from_account' => $fromAccount,
            'to_account' => $toAccount,
            'amount' => $amount,
            'asset_code' => $assetCode,
        ];

        if ($reference !== null) {
            $data['reference'] = $reference;
        }

        $response = $this->post('/transfers', $data);

        return new Transfer($response['data']);
    }

    /**
     * Get transfer details.
     *
     * @param  string  $transferId  Transfer UUID
     */
    public function get(string $transferId): Transfer
    {
        $response = $this->get("/transfers/{$transferId}");

        return new Transfer($response['data']);
    }

    /**
     * List transfers.
     *
     * @param  int  $page  Page number
     * @param  int  $perPage  Items per page
     */
    public function list(int $page = 1, int $perPage = 20): PaginatedResponse
    {
        $response = $this->get('/transfers', ['page' => $page, 'per_page' => $perPage]);

        return new PaginatedResponse($response, Transfer::class);
    }
}
